src/wasm-lib/kcl/src/execution/mod.rs

```diff
@@ -215,9 +215,8 @@ impl GlobalState {
             artifact_graph: Default::default(),
         };
 
-        // TODO(#4434): Use the top-level file's path.
-        let root_path = PathBuf::new();
         let root_id = ModuleId::default();
+        let root_path = settings.current_file.clone().unwrap_or_default();
```

Comment:
I was kind of hoping this would fix the order of the error message here to be input.kcl -> import_cycle2 -> import_cycle3 -> input.kcl

https://github.com/KittyCAD/modeling-app/blob/69fec37107137b621613c13a64f8ddaffcca2474/src/wasm-lib/kcl/tests/import_cycle1/execution_error.snap#L7-L9

Reply:
oh it would if i passed it thru the rust side haha ill do that

Reply:
basically just the test code doesnt route it thru but i will

Reply:
hmm it changed it but didnt do what you expected

Reply:
I had to add this here: https://github.com/KittyCAD/modeling-app/pull/5171/files#diff-5536303b6b946b693bb87aced2e48488068e25036236aa5793933d17eb93a005R2606

Reply:
That's close enough. Thanks!

Reply:
I'll link the issue so that this PR closes it too. #4434